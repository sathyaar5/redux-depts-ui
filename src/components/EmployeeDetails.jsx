import React from 'react';
import { Button, Table, TableHead, TableRow, TableCell, TableBody } from '@mui/material';
import { useDispatch, useSelector } from 'react-redux';
import { deleteEmployee } from '../actions';

const EmployeeDetails = ({ employees }) => {
  const dispatch = useDispatch();
  const departments = useSelector(state => state.departments);
  const selectedDepartmentIndex = useSelector(state => state.selectedDepartmentIndex);

  const handleDeleteEmployee = (employeeId) => {
    const department = departments[selectedDepartmentIndex];
    const employeeIndex = department.employeeDetails.findIndex(employee => employee.id === employeeId);
    if (employeeIndex !== -1) {
      dispatch(deleteEmployee(selectedDepartmentIndex, employeeIndex));
    }
  };

  return (
    <Table>
      <TableHead>
        <TableRow>
          <TableCell>ID</TableCell>
          <TableCell>Name</TableCell>
          <TableCell>Designation</TableCell>
          <TableCell>Branch</TableCell>
          <TableCell>Action</TableCell>
        </TableRow>
      </TableHead>
      <TableBody>
        {employees.map((employee) => (
          <TableRow key={employee.id}>
            <TableCell>{employee.id}</TableCell>
            <TableCell>{employee.name}</TableCell>
            <TableCell>{employee.designation}</TableCell>
            <TableCell>{employee.branch}</TableCell>
            <TableCell>
              <Button variant="contained" onClick={() => handleDeleteEmployee(employee.id)}>Delete</Button>
            </TableCell>
          </TableRow>
        ))}
      </TableBody>
    </Table>
  );
};

export default EmployeeDetails;