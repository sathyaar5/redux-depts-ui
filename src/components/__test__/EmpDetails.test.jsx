import React from "react";
import { render, screen } from "@testing-library/react";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";

import EmployeeDetails from "../EmployeeDetails";
import store from "../../store";

test("renders employee details correctly", () => {
  const employees = [
    { id: 1, name: "Sathya", designation: "Developer", branch: "Bengaluru" },
    {
      id: 2,
      name: "Mohammed Ibrahim ",
      designation: "Developer",
      branch: "Chennai",
    },
  ];

  const component = renderer.create(
    <Provider store={store}>
      <MemoryRouter>
        <EmployeeDetails employees={employees} />
      </MemoryRouter>
    </Provider>,
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
