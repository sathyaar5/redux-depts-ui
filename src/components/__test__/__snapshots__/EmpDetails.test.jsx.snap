// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders employee details correctly 1`] = `
<table
  className="MuiTable-root css-rqglhn-MuiTable-root"
  role={null}
>
  <thead
    className="MuiTableHead-root css-15wwp11-MuiTableHead-root"
    role={null}
  >
    <tr
      className="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
      role={null}
    >
      <th
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
        scope="col"
      >
        ID
      </th>
      <th
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
        scope="col"
      >
        Name
      </th>
      <th
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
        scope="col"
      >
        Designation
      </th>
      <th
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
        scope="col"
      >
        Branch
      </th>
      <th
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
        scope="col"
      >
        Action
      </th>
    </tr>
  </thead>
  <tbody
    className="MuiTableBody-root css-apqrd9-MuiTableBody-root"
    role={null}
  >
    <tr
      className="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
      role={null}
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
      >
        1
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
      >
        Sathya
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
      >
        Developer
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
      >
        Bengaluru
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-sghohy-MuiButtonBase-root-MuiButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          Delete
        </button>
      </td>
    </tr>
    <tr
      className="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
      role={null}
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
      >
        2
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
      >
        Mohammed Ibrahim 
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
      >
        Developer
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
      >
        Chennai
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-sghohy-MuiButtonBase-root-MuiButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          Delete
        </button>
      </td>
    </tr>
  </tbody>
</table>
`;
